x/evm/types/msg.go

```diff
@@ -56,90 +56,82 @@ const (
 
 // NewTx returns a reference to a new Ethereum transaction message.
 func NewTx(
-	chainID *big.Int, nonce uint64, to *common.Address, amount *big.Int,
-	gasLimit uint64, gasPrice, gasFeeCap, gasTipCap *big.Int, input []byte, accesses *ethtypes.AccessList,
+	tx *EvmTxArgs,
 ) *MsgEthereumTx {
-	return newMsgEthereumTx(chainID, nonce, to, amount, gasLimit, gasPrice, gasFeeCap, gasTipCap, input, accesses)
+	return newMsgEthereumTx(tx)
 }
 
 // NewTxContract returns a reference to a new Ethereum transaction
 // message designated for contract creation.
 func NewTxContract(
-	chainID *big.Int,
-	nonce uint64,
-	amount *big.Int,
-	gasLimit uint64,
-	gasPrice, gasFeeCap, gasTipCap *big.Int,
-	input []byte,
-	accesses *ethtypes.AccessList,
+	tx *EvmTxArgs,
```

Comment:
this is breaking since it changes the function signature. I suggest:

1. undo the changes and keep the `NewTx` and `NewTxContract` as is
2. Add a `Deprecated:` comment to mention that it's going to be deprecated in the future
2. Create a new constructor `NewEVMTx` that takes the tx args and use that function instead of the `NewTx` and `NewTxContract`

Reply:
the change to `newMsgEthereumTx` is ok since it's private

Reply:
Decided this is not relevant since its only test functions